File: 3/2.py
from functools import reduce
import operator

with open("input", "r") as file:
    text = file.read().splitlines()

limit = len(text[0])

steps = [
    (1, 1),
    (3, 1),
    (5, 1),
    (7, 1),
    (1, 2)
]

anss = []
for x, y in steps:
    ans = 0
    xcount = 0

    for line in text[y::y]:
        xcount += x
        current = line[xcount % limit]
        
        ans += 1 if current == "#" else 0

    anss.append(ans)

print(reduce(operator.mul, anss, 1))

File: 3/2.rs
use std::fs::read_to_string;

fn main() {
    let input = read_to_string("./input").unwrap();

    let vec_lines: Vec<_> = input.lines().collect();

    let limit = vec_lines[0].len();
    
    let steps = vec![
        (1, 1),
        (3, 1),
        (5, 1),
        (7, 1),
        (1, 2)
    ];

    let mut res: u32 = 1;

    for (x, y) in steps.iter() {
        let mut xcount = 0;
        let mut ans = 0;

        for line in vec_lines[*y as usize..].iter().step_by(*y) {
            xcount += x;
            let current = line.chars().nth(xcount % limit).unwrap();
    
            ans += if current == '#' {1} else {0};
        }
        res *= ans;
    }

    println!("{}", res);
}